### PR TITLE
feat(Wallet): sync saved addresses 7229

### LINKED ISSUE #7229: [Wallet] Sync saved addresses
This work package depends on work done to migrate add favourite to status-desktop and migrate mobile-favourites as a flag to saved addresses; see: https://github.com/status-im/status-desktop/issues/6546

### Description

Implement syncing of saved addresses in status-go`

### Acceptance Criteria

- [ ] Sync existing saved addresses across accounts
- [ ] Trigger syncing every time a change is done to saved addresses

### PR DESCRIPTION
**Pull request by stefandunca**
### Closes #7229

Requires status-go change: https://github.com/status-im/status-go/pull/2865

### Affected areas

Wallet - Saved Addresses
Device Sync

### Supported test cases

Existing saved address migration and full sync

- Existing paired devices that have different saved addresses should sync on a manual sync trigger

For paired devices

- any change on one side should be mirrored on the other when online
- doing conflicting changes on the same saved address (same address) are resolved by applying the whole state of the newest change
  - newest change is decided based on the global time of the system

### StatusQ checklist

- [x] ~~add documentation if necessary (new component, new feature)~~
- [x] ~~update sandbox app~~
- [x] ~~test changes in both light and dark theme?~~

### Demo

https://user-images.githubusercontent.com/47554641/193814010-56c9835a-4ab6-4643-bc94-753f9c6a4e3f.mov

### CODE REVIEW

**Comment by stefandunca** [comment]
@anastasiyaig please, test manual syncing of already paired devices, with an older version that has saved addresses, before switching to the new version (PR). This didn't work for some personal old test user folders and not sure if I should investigate it forward or the state were broken from debugging

**Review by borismelnik — APPROVED** [review]
LGTM 👍

**Review by alaibe — APPROVED** [review]

File: src/backend/backend.nim [hunk 1]
```diff
@@ -77,10 +77,11 @@ rpc(addEthereumChain, "wallet"):
 rpc(deleteEthereumChain, "wallet"):
   chainId: int
 
-rpc(addSavedAddress, "wallet"):
+rpc(addOrUpdateSavedAddress, "wakuext"):
```
[the remark this answers is not in the document]
Comment (PascalPrecht) [reply src/backend/backend.nim:80]: As you've mentioned, services APIs are just CRUD. They don't know about syncing.
There's two options

1) Use service APIs and manually trigger sync via another RPC call to messenger
2) Use messenger API that does syncing implicitly

When I was working on syncing bookmarks, I think @cammellos even mentioned that we probably want to move away from using service APIs altogether and always go through `Messenger` (notice that "Messenger" is not the best name anymore as it does many more things)

[the remark this answers is not in the document]
Comment (stefandunca) [reply src/backend/backend.nim:80]: Wallet APIs are just a wrapper over persistence (DB) so we still access them to fetch saved addresses and messenger module for syncing.
@alaibe I also didn't like that messenger has the responsibility of syncing logic and tried to look into refactoring syncing by separating it into a service that depends on messenger and persistance. However, I quickly realised that it is too early for me to do that and not many benefits for now. Still in the long run there are benefits in separating concerns of syncing and communication out of messenger.

**Comment by anastasiyaig** [comment]
i've tested that and i come up with 2 issues that are not caused by this PR:

1. the devices are visible for each other only after restart
2. for second instance of the app the back up seed is displayed

I will log them separately

**Review by anastasiyaig — APPROVED** [review]

**Pull request merged**